client/blocks/login/login-form.jsx

```diff
@@ -40,7 +40,25 @@ export class LoginForm extends Component {
 	};
 
 	componentDidMount() {
-		this.setState( { isDisabled: false } ); // eslint-disable-line react/no-did-mount-set-state
+		this.setState( { isDisabled: false }, () => { // eslint-disable-line react/no-did-mount-set-state
+			this.usernameOrEmail.focus();
+		} );
+	}
+
+	componentDidUpdate( prevProps ) {
+		const { requestError } = this.props;
+
+		if ( prevProps.requestError || ! requestError ) {
+			return;
+		}
+
+		if ( requestError.field === 'password' ) {
+			defer( () => this.password.focus() );
```

Comment:
I had to use `defer()` as otherwise fields would not focus (that's an hack found in https://github.com/facebook/react/issues/1791).